code/game/gamemodes/wizard/soulstone.dm

```diff
@@ -79,8 +79,33 @@
 	for(var/mob/living/simple_animal/shade/A in src)
 		A.death()
 	remove_held_body()
+	STOP_PROCESSING(SSobj, src)
 	return ..()
 
+/obj/item/soulstone/process()
+	. = ..()
+	if(held_body)
+		var/new_filter = isnull(get_filter("ray"))
+		if(!purified)
+			add_filter(name = "ray", priority = 1, params = list(
+				type = "rays",
+				size = 40,
+				color = "#c926ae",
+				factor = 6,
+				density = 20
+			))
```

Comment:
I think because we're adding this ray effect to more and more things, we need to either componentize this, or at least make some sort of helper proc to avoid duplicate code.